### ISSUE TITLE
NX-KS compatibility

### ISSUE THREAD

**Issue by KinoSeed** [mewlips/nx-remote-controller-mod#7]
I saw a request for the remote-controller to be added to nx-ks, and I have to say it looks pretty nice from what I can see without running it.

It will be best if we can manage the install in one (variable name) directory, so it can be easily and cleanly removed. 

so it installs "busybox" and it basically runs a Mongoose server from the camera and serves a page to the phone by a local network, the phone runs the script, commands get sent to the "deamon" which executes them?

since the memory mapping is different after you install "on-wake", we have to be careful with "poker", and if you do memory patching on addresses it may not work. (I saw only on 2 places, however they don't look like will be a problem)

I'm sort of a noob when it comes to linux :) so  your guidance is much appreciated.

**Comment by mewlips** [comment]
I modified installation scripts.
New release version is installable on NX-KS2 mod applied camera.
Nx-remote-controller-mod with NX-KS2 mod is working pretty well. :)

However, there is a problem related with /opt/home/scripts/auto folder.
nx-remote-controll-mod make a symbolic link in /opt/home/scripts/auto folder on installation process.
(/opt/usr/apps/nx-remote-controller-mod/nx-remote-controller-mod.sh --> /opt/home/scripts/auto/nx-remote-controller-mod.sh)
nx-remote-controller-mod.sh executes a daemon and related processes.

When reinstall (or upgrade) the nx-ks mod, this symbolic link will be removed.
Therefore, reinstalling the nx-remote-controller-mod is required after reinstalling the nx-ks mod.
This problem also exist on nx-patch with the same reason.

**Comment by KinoSeed** [comment]
If you don't mind I was thinking to pack it inside NX-KS, give it a separate button in the menu, so it will be ready and won't need separate install. 

If there are files installed outside "/opt/home/scripts/", uninstallation of NX-KS will need to take that into account, otherwise they will be left behind, which I like to avoid.

**Comment by mewlips** [comment]
nx-remote-controller-mod installs files to /opt/usr/apps/nx-remote-controller-mod.
The directory structure is based on @ottokiksmaler's proposals. (ottokiksmaler/nx500_nx1_modding#55)

And, free space of /opt partition is too small. (37MB free space on my nx500)
nx-remote-controller-mod uses 24M disk space.
It contains chroot environment that including libc and x-window libraries and binaries.

Therefore, if there are several standalone app, it is appropriate to be placed under the /opt/usr partition.
/opt/usr partition is almost not used and it has very big size. (2GB)

~~To pack the remote mod inside NX-KS is inappropriate, IMHO.
Standalone app should be installable independently from the master mod which has BT mod and keyscan.
Keyscan is the essential for the nx hacks. It is like the rc system in unix OS.~~

~~After installed the master mod which contains keyscan, then it should not be updated frequently.
When uninstalling the master mod, all of the standalone apps should be uninstalled also.~~

Edited:
@KinoSeed
If you think NX-KS is a collection of mods, I agree on your opinion.

**Comment by KinoSeed** [comment]
I kind of got it to work (from the sdcard for now) - the interface and control of the camera, but not the custom scripts.
seems like the problem I get is from initiating the named pipe, where mkfifo is not recognized as a command. Any suggestions? maybe "mknod /tmp/fifo p" ? (I'm not sure what I neet to cat there) 

You should be able to safely install nx-ks on a "wake-on moded camera", and take a look at it by removing "poker" first, and as long as you don't have another "keyscan" running, it should work on top of wake-on.  (modding resolutions and bitrates will not work of course)

Have you thought if it's possible to start the camera as a network-hotspot? 
But even now, as long as you pack the resources js/css/fonts with the control-page, you should be able to start your phone as a "hotspot" and connect the camera to it - eliminating the necessity of a network wifi.

**Comment by KinoSeed** [comment]
got some of the core functionality working:
https://www.facebook.com/NXKS2/posts/314584215584702

**Comment by mewlips** [comment]
Nice work! :-)

> seems like the problem I get is from initiating the named pipe, where mkfifo is not recognized as a command. Any suggestions? maybe "mknod /tmp/fifo p" ? (I'm not sure what I neet to cat there) 

If you do not include custom ui of rc mod, fifo is not necessary.
fifo is used for commuication with chrooted environment and outside.
Because I failed to build some necessary utilities (xdotool, yad, xev-nx ...) in native enviroment, I used the [buildroot](https://buildroot.org/) that build whole root file system and can run via chroot command.

FYI, mkfifo command can run via busybox.

```
chroot $TOOLS_PATH /bin/busybox mkfifo
```

> Have you thought if it's possible to start the camera as a network-hotspot?

Basically, camera as a network-hospot in nx system seems to be used for samsung's remote app.
So, it is hard to control without knowing the samsung's api(?).

> But even now, as long as you pack the resources js/css/fonts with the control-page, you should be able to start your phone as a "hotspot" and connect the camera to it - eliminating the necessity of a network wifi.

Great! I will merge these modifications in my repository.

**Comment by mewlips** [comment]
@KinoSeed I pushed a commit for support NX-KS2. 24047e532c36d5
https://github.com/mewlips/nx-remote-controller-mod/releases/download/v0.8.1/nx-ks2_nx-rc-0.8.1.zip

Would you test this version?

**Comment by KinoSeed** [comment]
Works great ! Thanks.
packing it as an update
